examples/morpheusvm/controller/controller.go

```diff
@@ -145,7 +142,7 @@ type Controller struct {
 
 func (c *Controller) Rules(t int64) chain.Rules {
 	// TODO: extend with [UpgradeBytes]
-	return c.genesis.Rules(t, c.networkID, c.chainID)
+	return c.genesis.GetRulesFactory().GetRules(t)
```

Comment:
Can we remove this from the controller interface so that the controller doesn't need to know about the genesis at all?

Ideally, the VM knows about the genesis/upgrade rules factory etc, and we remove this completely from the controller

Reply:
I was saving that for part 2 since this was going to be big.